test-data/unit/check-async-await.test

```diff
@@ -925,3 +925,21 @@ async def f() -> AsyncGenerator[int, None]:
 
 [builtins fixtures/async_await.pyi]
 [typing fixtures/typing-async.pyi]
+
+[case testAwaitUnion]
```

Comment:
Does this also fix type checking yield in generators? If yes, can you add a test case for this?

Reply:
Yeah it does, will do!